app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CreateEventActivity.kt

```diff
@@ -242,17 +283,63 @@ class CreateEventActivity : ActivityForAccessingTumOnline<CreateEventResponse>(R
         }
         event.description = description
         this.event = event
+        events = generateEvents()
 
-        val apiCall = apiClient.createEvent(event, null)
-        fetch(apiCall)
+        for (curEvent in events!!) {
+            val apiCall = apiClient.createEvent(curEvent, null)
+            fetch(apiCall)
```

Comment:
So what is supposed to happen when one of the requests fails? Will it fail everything? Will it try to complete the rest?

Reply:
Currently the ProgressActivity handles these errors and displays error messages/a retry dialog. If it's impossible to fetch all requests even after retrying, the CreateEventActivity would not automatically finish (Worst case: the user clicks on create again and events are created twice).

We could close the activity upon error and show a dialog advising the user to delete eventually created events and try again.

Reply:
Yea, so this retry behavior might be unexpected. With your addition of the new database column, you could insert the created entries one after each other, helping you to not create the same entry twice. Not sure that "advising" the user is the best UX :stuck_out_tongue:

Reply:
I'm not entirely sure what you mean, my concern is the following: 10 events should be created, the 6th, 7th.. fail 5 events are created in the app and uploaded to tum online. 
My preferred solution would be to delete all previously created events but chances are that the API problem is likely not resolved (eg. bad internet connection) now half of the entries are created and can not be deleted from the remote calendar. 
Is that the kind of problem you are thinking about or am I missing something else? :)

Reply:
I think we are on the same page. 

What I'm trying to hint at is: if you insert the entries into the database as you do each call, then the set of events you have in the DB with the same series ID, will let you do a more graceful "retry" command. This is however a lot of work and as you say, chances are, the connection is still bad.

So I'd suggest in a failure scenario to do the following: inform the user about the failure with "X out of Y events were created" and just close the dialog. This will give the user actionable intel about what they need to do to fix it and it makes the control flow easier to understand in this case.

Reply:
Done :) I simulated this by injecting an error, looks like this and appears to work fine:

![Screenshot_20201125-100455~2](https://user-images.githubusercontent.com/44805696/100206275-68d48700-2f06-11eb-8f53-8e40debbf7c7.jpg)